### PR TITLE
[Merged by Bors] - feat(smartmodule): added chain support to producer

### PR DESCRIPTION
**Pull request by galibey**
This PR is a step towards supporting `transform` section for all types of connectors. Specifically, here we add support of the chain of smartmodules to Fluvio Producer.
Main changes in this PR:
1. Decouple the creation and filling of the chain builder from the engine. Now it is required only in the instantiation phase. 
2. Fluvio Producer can be built with the chain builder passed from external.

### CODE REVIEW

File: crates/fluvio-smartengine/Cargo.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "fluvio-smartengine"
-version = "0.4.0"
+version = "0.4.1"
```
Comment (sehz) [inline crates/fluvio-smartengine/Cargo.toml:3]: since API is changed, we need to bump this to 0.5.0

File: crates/fluvio/Cargo.toml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "fluvio"
-version = "0.14.0"
```
Comment (sehz) [inline crates/fluvio/Cargo.toml:3]: need to bump to `0.15.0` with API change

**Review by sehz — CHANGES_REQUESTED** [review]
Looks very good.  only issue is with that we need to bump up version

File: crates/fluvio-smartengine/Cargo.toml (hunk 1 above)
Comment (galibey) [reply crates/fluvio-smartengine/Cargo.toml]: done

File: crates/fluvio/Cargo.toml (hunk 2 above)
Comment (galibey) [reply crates/fluvio/Cargo.toml:3]: done

**Review by sehz — APPROVED** [review]
LGTM

**Pull request closed without merging**